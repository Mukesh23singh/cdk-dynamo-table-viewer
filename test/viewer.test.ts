import { Stack, App } from '@aws-cdk/cdk';
import dynamodb = require('@aws-cdk/aws-dynamodb');
import { TableViewer } from '../lib';

test('happy  flow', () => {
  // GIVEN
  const app = new App();
  const stack = new Stack(app, 'test');
  const table = new dynamodb.Table(stack, 'MyTable', {
    partitionKey: { name: 'id', type: dynamodb.AttributeType.String }
  });

  // WHEN
  new TableViewer(stack, 'viewer', { table });

  // THEN
  expect(app.synthesizeStack('test').template).toMatchSnapshot();
});