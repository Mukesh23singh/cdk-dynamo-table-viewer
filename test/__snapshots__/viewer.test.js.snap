// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy  flow 1`] = `
Object {
  "Outputs": Object {
    "viewerViewerEndpoint2EEAF9D2": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "viewerViewerEndpointDED2DEA4",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "viewerViewerEndpointDeploymentStageprod95943DC5",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "viewerRenderedCodeArtifactHashA6A51A03": Object {
      "Description": "Artifact hash for asset \\"test/viewer/Rendered/Code\\"",
      "Type": "String",
    },
    "viewerRenderedCodeS3BucketF3062D53": Object {
      "Description": "S3 bucket for asset \\"test/viewer/Rendered/Code\\"",
      "Type": "String",
    },
    "viewerRenderedCodeS3VersionKey5D306085": Object {
      "Description": "S3 key for asset version \\"test/viewer/Rendered/Code\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyTable794EDED1": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
      },
      "Type": "AWS::DynamoDB::Table",
    },
    "viewerRenderedApiPermissionANYCDFA475E": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "viewerRenderedEEFA1E5E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "viewerViewerEndpointDED2DEA4",
              },
              "/",
              Object {
                "Ref": "viewerViewerEndpointDeploymentStageprod95943DC5",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "viewerRenderedApiPermissionANYproxyF60D17F7": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "viewerRenderedEEFA1E5E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "viewerViewerEndpointDED2DEA4",
              },
              "/",
              Object {
                "Ref": "viewerViewerEndpointDeploymentStageprod95943DC5",
              },
              "/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "viewerRenderedApiPermissionTestANYAB12F40E": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "viewerRenderedEEFA1E5E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "viewerViewerEndpointDED2DEA4",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "viewerRenderedApiPermissionTestANYproxy9F4CBAB6": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "viewerRenderedEEFA1E5E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "viewerViewerEndpointDED2DEA4",
              },
              "/test-invoke-stage/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "viewerRenderedEEFA1E5E": Object {
      "DependsOn": Array [
        "viewerRenderedServiceRoleDefaultPolicy196964DF",
        "viewerRenderedServiceRole7AE70B4C",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "viewerRenderedCodeS3BucketF3062D53",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "viewerRenderedCodeS3VersionKey5D306085",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "viewerRenderedCodeS3VersionKey5D306085",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": Object {
              "Ref": "MyTable794EDED1",
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "viewerRenderedServiceRole7AE70B4C",
            "Arn",
          ],
        },
        "Runtime": "nodejs8.10",
      },
      "Type": "AWS::Lambda::Function",
    },
    "viewerRenderedServiceRole7AE70B4C": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "lambda.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "viewerRenderedServiceRoleDefaultPolicy196964DF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "MyTable794EDED1",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "viewerRenderedServiceRoleDefaultPolicy196964DF",
        "Roles": Array [
          Object {
            "Ref": "viewerRenderedServiceRole7AE70B4C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "viewerViewerEndpointANY4A9252D2": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "viewerRenderedEEFA1E5E",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "viewerViewerEndpointDED2DEA4",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "viewerViewerEndpointDED2DEA4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "viewerViewerEndpointAccount1D965397": Object {
      "DependsOn": Array [
        "viewerViewerEndpointDED2DEA4",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "viewerViewerEndpointCloudWatchRole28045685",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "viewerViewerEndpointCloudWatchRole28045685": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "apigateway.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "viewerViewerEndpointDED2DEA4": Object {
      "Properties": Object {
        "Name": "ViewerEndpoint",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "viewerViewerEndpointDeployment00B06945eaab38fe899ff2d44175a541c249d91c": Object {
      "DependsOn": Array [
        "viewerViewerEndpointproxyANYF7759A2A",
        "viewerViewerEndpointproxy103C52F5",
        "viewerViewerEndpointANY4A9252D2",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "viewerViewerEndpointDED2DEA4",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "viewerViewerEndpointDeploymentStageprod95943DC5": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "viewerViewerEndpointDeployment00B06945eaab38fe899ff2d44175a541c249d91c",
        },
        "RestApiId": Object {
          "Ref": "viewerViewerEndpointDED2DEA4",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "viewerViewerEndpointproxy103C52F5": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "viewerViewerEndpointDED2DEA4",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "viewerViewerEndpointDED2DEA4",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "viewerViewerEndpointproxyANYF7759A2A": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "viewerRenderedEEFA1E5E",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "viewerViewerEndpointproxy103C52F5",
        },
        "RestApiId": Object {
          "Ref": "viewerViewerEndpointDED2DEA4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
  },
}
`;
